polygeist/lib/Dialect/Polygeist/IR/PolygeistOps.cpp

```diff
@@ -36,6 +36,26 @@ using namespace mlir::arith;
 llvm::cl::opt<bool> BarrierOpt("barrier-opt", llvm::cl::init(true),
                                llvm::cl::desc("Optimize barriers"));
 
+/// Return \p mrTy's address space as an integer (being 0 the default) if it is
+/// represented as an integer attribute.
+static std::optional<unsigned> getMemorySpaceAsIntIfInteger(MemRefType mrTy) {
+  Attribute memorySpace = mrTy.getMemorySpace();
+  if (!memorySpace)
+    return 0;
```

Comment:
Use enumerator for global address space value ?

Reply:
This function is intended to return the address space as an integer, not an enum. This follows MLIR's convention ([see](https://github.com/llvm/llvm-project/blob/75501f53624de92aafce2f1da698b249a7293dc7/mlir/lib/IR/BuiltinTypes.cpp#L510)).

Reply:
Also, this is not the global address space, this is not a SYCL concept, it's just a means to translate an attribute to an integer.

Reply:
Ok then we aren't deviating from LLVM proper, which is good.  We could return the enumerator as an integer after a static_cast but is fine to use zero given that LLVM is not doing any better.

Reply:
what happens when we use `sycl.access.address_space` ?

Reply:
- Verification: verifies, as it is not an `IntegerAttr`
- Conversion: checks whether the address space of the pointer and the `sycl.access.address_space` after being converted to the numeric representation for the target match